classes/migration/upgrade/OJSv3_3_0UpgradeMigration.inc.php

```diff
@@ -41,4 +43,114 @@ public function down() {
 			$table->text('setting_value')->nullable()->change();
 		});
 	}
+
+	/**
+	 * @return void
+	 * @bried reset serialized arrays and convert array and objects to JSON for serialization, see pkp/pkp-lib#5772
+	 */
+	private function _settingsAsJSON() {
+
+		// Convert settings where type can be retrieved from schema.json
+		$schemaDAOs = ['SiteDAO', 'AnnouncementDAO', 'AuthorDAO', 'ArticleGalleyDAO', 'JournalDAO', 'EmailTemplateDAO', 'PublicationDAO', 'SubmissionDAO'];
+		$processedTables = [];
+		foreach ($schemaDAOs as $daoName) {
+			$dao = DAORegistry::getDAO($daoName);
+			$schemaService = Services::get('schema');
+			$schema = $schemaService->get($dao->schemaName);
+			$tableName = $dao->settingsTableName;
+			$processedTables[] = $tableName;
+			foreach ($schema->properties as $propName => $propSchema) {
+				if (empty($propSchema->readOnly)) {
+					if ($propSchema->type === 'array' || $propSchema->type === 'object') {
+						Capsule::table($tableName)->where('setting_name', $propName)->get()->each(function ($row) use ($tableName, $propName) {
+							$this->_toJSON($row, $tableName, ['setting_name', 'locale'], 'setting_value');
+						});
+					}
+				}
+			}
+		}
+
+		// Convert settings where only setting_type column is available
+		$tables = Capsule::connection()->select('SHOW TABLES');
```

Comment:
Have you tried this with PostgreSQL? I don't think PostgreSQL supports `SHOW TABLES`.

Reply:
(I don't think there's any harm in just listing the affected tables, as you've done above for other cases; it makes things more greppable.)

Reply:
Indeed, doesn't support it.